### PR TITLE
Documentation: Add documentation on time series causal graphs

### PR DESCRIPTION
**Pull request by francesca-raimondi-causalens**
## Description and Motivation
Add documentation on time series causal graphs

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactor (moving code around, general refactoring improvements)
- [x] Documentation (documentation)

## PR Checklist:

- [x] I have reviewed my own PR? (review the PR as you are reviewing someone else's PR)
- [x] I have implemented all requirements? (see JIRA, project documentation)
- [ ] I am affecting someone else's work? If so: I have added those people as reviewers?
- [x] I have added comments to all the bits that are hard to follow
- [ ] At a bare minimum, I tested this manually
- [ ] I have added unit test(s)
- [ ] Is this a bugfix? If so have I added a regression test?
- [x] Does this PR satisfy the [one PR, one concern](https://medium.com/@fagnerbrack/one-pull-request-one-concern-e84a27dfe9f1) rule?
- [x] If needed, documentation has been added/updated?
- [ ] I have updated the appropriate changelog with a line for my changes
- [ ] If this PR breaks reproducibility, have I added a note in the changelog explaining the break in reproducibility
      and its extent?

### CODE REVIEW

**Comment by andrew-causalens** [comment]
@francesca-raimondi-causalens please resolve conflicts and generate images using Dara as we discussed on Slack.

File: docs/images/ts_minimal_graph.png
Comment (andrew-causalens) [inline docs/images/ts_minimal_graph.png:1]: Would align nodes at same time stamp but t-1 and t are not aligned. Should match image format from Elements of Causal Inference textbook.

File: docs/images/ts_cg.png
Comment (andrew-causalens) [inline docs/images/ts_cg.png:1]: Would align nodes at same time stamp but t-2, t-1, and t are not aligned. Should match image format from Elements of Causal Inference textbook.

File: docs/quickstart.md
Comment (andrew-causalens) [inline docs/quickstart.md:106]: This comment is not exactly correct as we don't add lagged nodes here. They are added automatically when the edges are added.

File: docs/time_series_causal_graph.md
Comment (andrew-causalens) [inline docs/time_series_causal_graph.md:349]: fonctionalities -> functionalities

File: docs/time_series_causal_graph.md
Comment (andrew-causalens) [inline docs/time_series_causal_graph.md:344]: Should not be `maxlag`

File: docs/time_series_causal_graph.md
Comment (andrew-causalens) [inline docs/time_series_causal_graph.md:5]: theotetical -> theoretical

File: docs/time_series_causal_graph.md
Comment (andrew-causalens) [inline docs/time_series_causal_graph.md:196]: Missing parameters.

File: docs/time_series_causal_graph.md
Comment (andrew-causalens) [inline docs/time_series_causal_graph.md:183]: Need an image of the extended graph.

File: docs/time_series_causal_graph.md
Comment (andrew-causalens) [inline docs/time_series_causal_graph.md:52]: Missing `EdgeType` import

File: docs/time_series_causal_graph.md
Comment (andrew-causalens) [inline docs/time_series_causal_graph.md:55]: Renamed `EDGE_T` to `EdgeType` in 0.2.0. Need to fix throughout.

File: docs/time_series_causal_graph.md
Comment (andrew-causalens) [inline docs/time_series_causal_graph.md:247]: I would not do this actually.  We fill this automatically from the identifier. Would hide how we use meta to keep this info away from the user.

File: docs/time_series_causal_graph.md
Comment (andrew-causalens) [inline docs/time_series_causal_graph.md:224]: Should be plural name

File: docs/time_series_causal_graph.md
Comment (andrew-causalens) [inline docs/time_series_causal_graph.md:221]: Would add type hinting as this is `TimeSeriesNode`

File: docs/time_series_causal_graph.md
Comment (andrew-causalens) [inline docs/time_series_causal_graph.md:153]: Should say this this returns true.

File: docs/time_series_causal_graph.md
Comment (andrew-causalens) [inline docs/time_series_causal_graph.md:159]: Full link to `CausalGraph`. Check throughout

File: docs/time_series_causal_graph.md
Comment (andrew-causalens) [inline docs/time_series_causal_graph.md:166]: Suggest actually adding type hint here. Would do throughout.

File: docs/time_series_causal_graph.md
Comment (francesca-raimondi-causalens) [reply docs/time_series_causal_graph.md:154]: As we don't know the return value, I added
`Returns True if the graph is a minimal graph, False otherwise.`

File: docs/images/ts_cg.png
Comment (andrew-causalens) [inline docs/images/ts_cg.png:1]: Not for this PR, but when you generate the updated images, please generate one in light mode and one in dark mode as I will use some to update the homepage of the CG website. Instructions are [here](https://docs.causalens.com/docs/_generated/dara-internal/docs/packages/dara-jupyter/introduction#adding-plugins).

File: docs/time_series_causal_graph.md
Comment (andrew-causalens) [reply docs/time_series_causal_graph.md:154]: Ah sorry, I thought you were asking if the `minimal_graph` was minimal which would obviously be `True`. Missed that it was the original graph. Would show that `minimal_graph` is `True` though.

File: docs/time_series_causal_graph.md
Comment (andrew-causalens) [reply docs/time_series_causal_graph.md:248]: This was not resolved but marked as resolved. Reopening.

File: docs/time_series_causal_graph.md
Comment (andrew-causalens) [inline docs/time_series_causal_graph.md:53]: Can be imported `from cai_causal_graph`. Put on same line as above. Fix throughout.

File: docs/time_series_causal_graph.md
Comment (andrew-causalens) [inline docs/time_series_causal_graph.md:67]: Missing `EdgeType`. Please double check throughout.

File: docs/time_series_causal_graph.md
Comment (andrew-causalens) [inline docs/time_series_causal_graph.md:1]: This page only represents DAGs. Do we want to show we support CPDAGs, MAGs, and PAGs as well? This functionality is currently broken but @matteo-causalens is working to fix as part of the UAT feedback.

Comment (andrew-causalens) [inline docs/time_series_causal_graph.md:1]: This is also a new page, but `sidebar.json` has not been updated.

Comment (andrew-causalens) [reply docs/time_series_causal_graph.md:1]: This is marked resolved again but not actually committed. Makes reviewing hard. Suggest pushing around same time as closing these comments.

Comment (andrew-causalens) [reply docs/time_series_causal_graph.md:1]: @matteo-causalens any opinion on this comment?

Comment (matteo-causalens) [reply docs/time_series_causal_graph.md:1]: I don't know honestly. In my opinion is an almost trivial extension as the main point for the user is to understand the concept of a TS graph. Maybe I would just add add example with a CPDAG and explain why it may be interesting. Because I think the user could be confused to see e.g. a bi-directed graph between X-2 and X for example. May be worth just to show an example and explain why this could happen

Comment (francesca-raimondi-causalens) [reply docs/time_series_causal_graph.md:1]: I have added a small example but I am unsure of the explanation that we should give, so I'll open a new conversation there.

File: docs/time_series_causal_graph.md
Comment (francesca-raimondi-causalens) [inline docs/time_series_causal_graph.md:218]: @andrew-causalens What explanation should we give here?

Comment (andrew-causalens) [reply docs/time_series_causal_graph.md:218]: That they can encode Markov equivalence class of time series causal graphs. It is not just CPDAGs but MAGs and PAGs. Would rename the heading to Markov Equivalence Classes and then use a CPDAG as the example. But mention one could create a MAG or PAG too.

**Comment by andrew-causalens** [comment]
Thanks @francesca-raimondi-causalens for making the requested changes. @matteo-causalens can you please take a review as you developed the `TimeSeriesCausalGraph` class initially? Thank you 🙏 

We have branch protection for two approvers on this repo so worth us both reviewing / approving before we merge.

**Comment by igorcausalens** [comment]
🔥

File: docs/time_series_causal_graph.md
Comment (matteo-causalens) [inline docs/time_series_causal_graph.md:1]: If you want to represent a minimal graph, I can't comment on the image `docs/images/ts_minimal_graph.png` but doesn't seem correct. The edge `X1 lag(n=1) -> X2 lag(n=1)` should not be there.

If the purpose was different, please ignore my comment.

**Review by matteo-causalens — COMMENTED** [review]
Great stuff @francesca-raimondi-causalens. Thank you for helping with TSCG. I just have a comment on the image for the minimal graph. If the comment is not correct, please let me know and I will approve.

**Review by matteo-causalens — APPROVED** [review]

File: docs/time_series_causal_graph.md
Comment (francesca-raimondi-causalens) [inline docs/time_series_causal_graph.md:383]: Reviewing this now.

File: docs/time_series_causal_graph.md
Comment (francesca-raimondi-causalens) [inline docs/time_series_causal_graph.md:404]: @matteo-causalens Could you add a line explaining how info is propagated from `t=0`?

File: docs/time_series_causal_graph.md
Comment (andrew-causalens) [inline docs/time_series_causal_graph.md:7]: Incorrect link throughout. It is `cai_causal_graph.time_series_causal_graph.TimeSeriesCausalGraph`. Same for `Node` and `TimeSeriesNode`.

File: docs/time_series_causal_graph.md
Comment (andrew-causalens) [inline docs/time_series_causal_graph.md:134]: More edges here than 1s in adjacency matrix. Will just remove.

**Review by andrew-causalens — APPROVED** [review]

**Pull request merged**
